server/src/main/java/com/skillmagnet/Course/CourseController.java

```diff
@@ -20,75 +20,82 @@ public class CourseController {
 
     /**
      * Retrieves specific course by id
+     *
      * @param id - id of course to find
      * @return - course object || NOT_FOUND
      */
     @GetMapping("/course/{id}")
-    public ResponseEntity<Course> getCourse(@PathVariable("id") int id){
+    public ResponseEntity<Course> getCourse(@PathVariable("id") int id) {
         Course course = courseRepository.findById(id);
 
-        if(course == null){
+        if (course == null) {
             return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
         }
         return new ResponseEntity<>(course, HttpStatus.OK);
     }
 
     /**
-     * 
-     * @return - all courses in course table in a list 
+     * @return - all courses in course table in a list
      */
     @GetMapping("/course")
-    public ResponseEntity<List<Course>> getAllCourses(){        
+    public ResponseEntity<List<Course>> getAllCourses() {
         return new ResponseEntity<>(courseRepository.findAll(), HttpStatus.OK);
     }
 
     /**
      * Create operation for course
-     * @param course - request body of json course object 
-     * 
-     * IE: 
-        {
-        "title": "First Course - Updated!",
-        "description": "Testing out new api - Updated!",
-        "category": "Testing - Updated!",
-        "views": 0,
-        "thumbsUp": 0,
-        "thumbsDown": 0
-        }
-     * @return
+     *
+     * @param course - request body of json course object
+     *               <p>
+     *               IE:
+     *               {
+     *               "title": "First Course - Updated!",
+     *               "description": "Testing out new api - Updated!",
+     *               "category": "Testing - Updated!",
+     *               "views": 0,
+     *               "thumbsUp": 0,
+     *               "thumbsDown": 0
+     *               }
+     * @return course object || bad request for missing title, category, and/or description
      */
     @PostMapping("/course")
-    public ResponseEntity<Course> createCourse(@RequestBody Course course){
+    public ResponseEntity<?> createCourse(@RequestBody Course course) {
+        if (null == course.getTitle() || null == course.getCategory() || null == course.getDescription()) {
+            return new ResponseEntity<>("Course does not have all required fields", HttpStatus.BAD_REQUEST);
+        }
+
         courseRepository.save(course);
         return new ResponseEntity<>(course, HttpStatus.OK);
     }
```

Comment:
:memo: Added check to make sure title, category, and description were not null. The other properties autogenerate or default to 0